node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/FailedExpirer.java

```diff
@@ -76,7 +76,7 @@ protected double maintain() {
                                          isExpired,
                                          (node, lock) -> recycle(node, List.of(), allNodes).get());
 
-        nodeRepository.nodes().performOnRecursively(allNodes.nodeType(NodeType.host),
+        nodeRepository.nodes().performOnRecursively(allNodes.nodeType(NodeType.host).matching(isExpired),
```

Comment:
Otherwise takes the tenant-host and unallocated-lock for each deprovisioned host ...